test/python/transpiler/test_dynamical_decoupling.py

```diff
@@ -224,15 +224,15 @@ def test_insert_dd_ghz_everywhere(self):
         expected = self.ghz4.copy()
         expected = expected.compose(Delay(50), [1], front=True)
 
-        expected = expected.compose(Delay(162), [2], front=True)
+        expected = expected.compose(Delay(163), [2], front=True)
         expected = expected.compose(YGate(), [2], front=True)
-        expected = expected.compose(Delay(326), [2], front=True)
+        expected = expected.compose(Delay(325), [2], front=True)
         expected = expected.compose(YGate(), [2], front=True)
         expected = expected.compose(Delay(162), [2], front=True)
 
-        expected = expected.compose(Delay(212), [3], front=True)
+        expected = expected.compose(Delay(213), [3], front=True)
         expected = expected.compose(YGate(), [3], front=True)
-        expected = expected.compose(Delay(426), [3], front=True)
+        expected = expected.compose(Delay(425), [3], front=True)
         expected = expected.compose(YGate(), [3], front=True)
         expected = expected.compose(Delay(212), [3], front=True)
```

Comment:
This is correct because it places the circuit in `front=True` (so order is reversed). I'm not sure why the reference circuit is implemented like this.

It is possible to keep original behavior, but as we have discussed, consideration of the alignment yields larger extra slack ~ 10s dt (previously this is up to 1 dt of rounding error). If we add this slack to the middle as it is, it will hurt the balance of the delay. To minimize the imbalance, the slack is now split into two edges. But this behavior can be controlled with the pass option, such as `extra_slack_distribution` in `["balanced", "middle"]`.

Reply:
hmm i have no idea why i used front=True :)

but where is the alignment constraint in this test that is causing the extra slack? in this test which is `test_insert_dd_ghz_everywhere` I see no alignment constraint, am I missing something?

```py
        pm = PassManager(
            [
            [
                ALAPSchedule(self.durations),
                ALAPSchedule(self.durations),
                DynamicalDecoupling(self.durations, dd_sequence, skip_reset_qubits=False),
                DynamicalDecoupling(self.durations, dd_sequence, skip_reset_qubits=False),
            ]
            ]
        )
        )
```

Reply:
There is no alignment constraint but finite rounding error. Extra slack comes from both rounding error (i.e. dt is integer) and alignment constraint.

Reply:
ok i see. yes i think an extra_slack_distribution argument could be nice

Reply:
Done in https://github.com/Qiskit/qiskit-terra/pull/7760/commits/7c27a85e53a31b1c3342cccc18fd7756051b6a5a